### PR TITLE
Vue2 add bounce, shake, and some tests

### PR DESCRIPTION
**Pull request by jasonlundien**
PR adds bounce, shake, beat-fade and tests in vue-fontawesome 2.x.
Also updates the deps (thank you @ej2)!

### CODE REVIEW

File: src/components/__tests__/FontAwesomeIcon.test.js
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js:44]: fix failing test here. `xmark` is the icon name, `close` is the alias.

**Review by ej2 — APPROVED** [review]
Everything looks great! I think @robmadole should review before we merge.

File: DEVELOPMENT.md
Comment (robmadole) [inline DEVELOPMENT.md]: @jasonlundien I've looked at a few projects and there doesn't seem to be a solid convention for how you tag major versions.

So let's do it like this:
- If it's compatible with Vue 2: `latest-2`
- For Vue 3: `latest-3`

File: package.json
Comment (robmadole) [inline package.json]: @jasonlundien @ej2 can we update the tests so that it works like our React component?

https://github.com/FortAwesome/react-fontawesome/blob/master/package.json#L38-L40

The reason for both of them is so that we can test with the FA 5 and 6 versions.

File: src/components/FontAwesomeIcon.js
Comment (robmadole) [inline src/components/FontAwesomeIcon.js:61]: I see what you did there...that's clever. Nice.

This seems rather tricky now. Have we documented how this works?

File: src/components/__tests__/FontAwesomeIcon.test.js
Comment (robmadole) [inline src/components/__tests__/FontAwesomeIcon.test.js:191]: I think if you add `expect(vm.$el.classList.contains('fa-flip')).toBeFalsy()` you will get a surprising test failure...I will comment down below as to why I think this.

File: src/utils.js
Comment (robmadole) [inline src/utils.js]: This probably needs to be
```suggestion
    'fa-flip': props.flip === true,
```

Otherwise since a string is truthy I think anytime the value is `horizontal` or `vertical` or `both` you'll get the `fa-flip` class applied.

**Review by robmadole — CHANGES_REQUESTED** [review]

File: src/components/FontAwesomeIcon.js
Comment (jasonlundien) [reply src/components/FontAwesomeIcon.js:61]: What is interesting about flip is that we use `flip` as a standalone animation that actually continuously flips the icon over and over.  Ex:  `<i class="fa-solid fa-compact-disc fa-flip"></i>`

And we also use `flip` to rotate icons (not an animation it just rotates the icon, not continuously).  Ex.  `<i class="fa-solid fa-snowboarding fa-flip-vertical"></i>`

We do have both documented:
Rotate: https://fontawesome.com/docs/web/style/rotate#rotate-classes
Animation: https://fontawesome.com/docs/web/style/animate#flip

The flip animation was NOT working with our Vue 2 or Vue 3 packages... this fixes it.

File: src/utils.js
Comment (jasonlundien) [reply src/utils.js]: You were/are right about the failing test @robmadole without having this... I actually saw my icons continuously rotating and ignored it for whatever reason.  This addition, however, does prevent the icons from flipping yet allowing them to have been "rotated" as desired.

**Review by robmadole — APPROVED** [review]

**Pull request merged**
